File: Data Structures & Algorithms/partition-labels/submission-0.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        lengthMap = {}
        result = []
        for i in range(len(s)):
            lengthMap[s[i]] = i
        size = end = 0
        for i in range(len(s)):
            size +=1
            end = max(end, lengthMap[s[i]])
            if i==end:
                result.append(size)
                size = 0
        return result

        

File: Data Structures & Algorithms/pow-x-n/submission-0.java
class Solution {
    public double myPow(double x, int n) {
        double result = 1.0;
        if(n<0){
            n = -n;
            x = 1/x;
        }
        while(n>0){
            if(n%2 != 0) result *= x;
            x *= x;
            n = n/2;
        }
        return result;
    }
}
